### PR TITLE
Basic framework

### PR DESCRIPTION
**Pull request by ZuhairGhias**
#18

### CODE REVIEW

File: Source/Assets/Weapon.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private LayerMask hitLayerMask;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        DebugUtility.HandleEmptyLayerMask(hitLayerMask, this, "Enemy/Floor/Walls");
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (AMastryukov) [inline Source/Assets/Weapon.cs:25]: Remove these bro

File: Source/Assets/Weapon.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    {
+
+    }
+
+    void TryFire()
```
Comment (AMastryukov) [inline Source/Assets/Weapon.cs:30]: should be private by default?

File: Source/Assets/Weapon.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        Transform cameraTransform = Camera.main.transform;
+        Debug.Log("FIRE");
+
+        if (Physics.Raycast(cameraTransform.position, cameraTransform.forward, out rayHit, range, hitLayerMask))
+        {
+            hitEnemy = rayHit.collider.gameObject.GetComponent<Enemy>();
+        }
+
+
+        if (hitEnemy != null)
```
Comment (AMastryukov) [inline Source/Assets/Weapon.cs:43]: This should go inside the Raycast if, no?

File: Source/Assets/Weapon.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+public class Weapon : Item
+{
+    [SerializeField]
+    private float spread;
+    [SerializeField]
+    private float damage;
+    [SerializeField]
+    private float range = 10f;
+    [SerializeField]
+    private AnimationCurve damageFallOff;
```
[the remark this answers is not in the document]
Comment (ZuhairGhias) [reply Source/Assets/Weapon.cs:14]: Ignore this. It will change a lot over the next day.

File: Source/Assets/Weapon.cs (hunk 3 above)
Comment (ZuhairGhias) [reply Source/Assets/Weapon.cs:43]: Written this way for Debug.DrawLine()

File: Source/Assets/Scripts/Objects/Item.cs [hunk 5]
```diff
@@ -3,7 +3,7 @@
 using UnityEngine.Events;
 public abstract class Item : MonoBehaviour
 {
-    public Action<string> StatusChanged;
+    public static Action<string> StatusChanged;
```
Comment (AMastryukov) [inline Source/Assets/Scripts/Objects/Item.cs:6]: OnStatusChanged

**Review by AMastryukov — APPROVED** [review]

**Pull request merged**